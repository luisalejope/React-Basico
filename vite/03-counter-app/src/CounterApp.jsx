import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value)

    const handleAdd = () => setCounter(counter + 1)
    

    const handleSubstract = () => {
        if (counter > 0) {
            setCounter(counter - 1)
        }
    }

    const handleReset = () => setCounter(value)

    return (
        <>
            <h1>CounterAPP</h1>
            <h2>{counter}</h2>
            <button aria-label='btn-add' onClick={handleAdd}>+1</button>
            <button aria-label='btn-reset' onClick={handleReset}>reset</button>
            <button aria-label='btn-rest' onClick={handleSubstract}>-1</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}