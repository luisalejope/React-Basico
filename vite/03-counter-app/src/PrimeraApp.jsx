import PropTypes from 'prop-types';
// functional component

export const PrimeraApp = ({ saludo, subtitulo }) => {
  /* const persona = {
    nombre: "Luis",
    apellido: "Peña",
    edad: "45",
    clave: "ironman",
    direccion: {
      ciudad: "Bogotá",
      zip: "1222231",
      lat: "14.53242",
      long: "32.131",
    },
  }; */

  return (
    <>
      <h1 data-testid='test-saludo'>{saludo}</h1>
      {/* <pre> {JSON.stringify(persona, null, 3)} </pre> */}
      <p>{subtitulo}</p>
    </>
  );
};
// Obliga los argumentos y el tipo
PrimeraApp.propTypes ={
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string
}
// Define las propiedades por defecto
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}
