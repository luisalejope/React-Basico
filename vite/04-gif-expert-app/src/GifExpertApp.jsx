import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch'])

    const onAddCategory = (newCategory) => {

        const alreadyExists = categories.includes(newCategory)

        if (!alreadyExists) {
            setCategories([newCategory, ...categories])
        }
    }

    return (
        <>
            <div>GifExpertApp</div>

            <AddCategory onAddCategory={onAddCategory} />

            <ol key={'hola'} className="categories">
                {categories.map(category => (
                    <GifGrid category={category} key={category} />
                ))}
            </ol>
        </>
    )
}
