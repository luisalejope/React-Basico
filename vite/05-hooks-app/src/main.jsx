import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
// import { HooksApp } from './HooksApp'
// import './08-useReducer/intro-reducer';
import { RouterProvider } from 'react-router-dom';
import { getRoutes } from './09-useContext/Routes';

const router = getRoutes()

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>,
)
