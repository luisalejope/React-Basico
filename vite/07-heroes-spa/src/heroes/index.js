
export * from './components'
export * from '../assets/data'
export * from './helpers'
export * from './HeroesApp'
export * from './pages'