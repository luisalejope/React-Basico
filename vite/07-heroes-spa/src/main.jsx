import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { AppRouter } from './router'
import { AuthProvider } from './auth'


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  </React.StrictMode>,
)
