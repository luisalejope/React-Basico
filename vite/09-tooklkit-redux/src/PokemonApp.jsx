import PropTypes from 'prop-types'
import { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getPokemons } from './store/slices/pokemon';

export const PokemonApp = () => {
    const {page, pokemons, isLoading} = useSelector(state => state.pokemons)
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(getPokemons());
    }, [])

    const onNextPage = () => {
        dispatch(getPokemons(page))
    }

    return (
        <>
            <h1>PokemonApp</h1>
            <hr />
            <span>Loading: {JSON.stringify(isLoading) }</span>

            <ul style={{display: !isLoading? 'block' : 'none'}}>
                {
                    pokemons.map(pokemon => (
                        <li key={pokemon.name}>{pokemon.name}</li>
                    ))
                }
            </ul>

            <p>page: {page}</p>
            <button disabled={isLoading} onClick={onNextPage}>Next page</button>
        </>
    )
}

PokemonApp.propTypes = {}

export default PokemonApp