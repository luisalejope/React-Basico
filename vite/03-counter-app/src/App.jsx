// import { CounterApp } from './CounterApp'
import {PrimeraApp} from './PrimeraApp'

export const App = () => {
  return (
    <>
      {/* <CounterApp value={1} /> */}
      <PrimeraApp saludo='Hola'/>
    </>
  )
}
