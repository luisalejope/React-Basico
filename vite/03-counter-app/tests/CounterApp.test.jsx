import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en <CounterApp />', () => {
    //1. Inicializar
    const initialValue = 100

    test('Debe hacer match con el snapshot', () => {
        //2. Estimulo
        const { container } = render(<CounterApp value={initialValue} />);
        //3. Observar Comportamiento
        expect(container).toMatchSnapshot()
    })

    test(`debe de mostrar el valor inicial de "${initialValue}"`, () => {
        //2. Estimulo
        render(<CounterApp value={initialValue} />);
        //3. Observar Comportamiento
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(initialValue.toString())
    })

    test('debe de incrementar con el boton +1', () => {
        //2. Estimulo
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('+1'))

        //3. Observar Comportamiento
        expect(screen.getByText('101')).toBeTruthy()
    })

    test('debe de decrementar con el boton -1', () => {
        //2. Estimulo
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('-1'))

        //3. Observar Comportamiento
        expect(screen.getByText('99')).toBeTruthy()
    })

    test('debe de funcionar el boton de reset', () => {
        //2. Estimulo
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        /* fireEvent.click(screen.getByText('reset')); */
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}))

        //3. Observar Comportamiento
        expect(screen.getByText(initialValue.toString())).toBeTruthy();
    })
})