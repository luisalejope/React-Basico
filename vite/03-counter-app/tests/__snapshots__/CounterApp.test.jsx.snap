// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterAPP
  </h1>
  <h2>
    100
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
  <button
    aria-label="btn-rest"
  >
    -1
  </button>
</div>
`;
